File: app/src/main/java/com/veyndan/paper/reddit/api/imgur/network/ImgurService.java
package com.veyndan.paper.reddit.api.imgur.network;

import android.support.annotation.NonNull;

import com.veyndan.paper.reddit.api.imgur.model.Album;
import com.veyndan.paper.reddit.api.imgur.model.Basic;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface ImgurService {

    @NonNull
    @GET("album/{id}")
    Single<Basic<Album>> album(@NonNull @Path("id") String id);
}

File: app/src/main/java/com/veyndan/paper/reddit/api/imgur/network/ImgurService.kt
package com.veyndan.paper.reddit.api.imgur.network

import com.veyndan.paper.reddit.api.imgur.model.Album
import com.veyndan.paper.reddit.api.imgur.model.Basic
import io.reactivex.Single
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Path

interface ImgurService {

    @GET("album/{id}")
    fun album(@Path("id") id: String): Single<Response<Basic<Album>>>
}
